pennylane/gradients/hadamard_gradient.py

```diff
@@ -298,8 +276,11 @@ def hadamard_grad(
     if argnum is None and not tape.trainable_params:
         return _no_trainable_grad(tape)
 
-    trainable_params = choose_trainable_params(tape, argnum)
-    diff_methods = find_and_validate_gradient_methods(tape, "analytic", trainable_params)
+    trainable_param_indices = choose_trainable_params(tape, argnum)
+    diff_methods = {
+        idx: _try_zero_grad_from_graph_or_get_grad_method(tape, tape.trainable_params[idx], True)
```

Comment:
We no longer use `Operator.grad_method` to say whether or not we can do hadamard grad with it. We just care if there's a generator.